pkg/render/node.go

```diff
@@ -491,11 +491,13 @@ func (c *nodeComponent) nodeDaemonset(cniCfgMap *v1.ConfigMap) *apps.DaemonSet {
 
 // nodeTolerations creates the node's tolerations.
 func (c *nodeComponent) nodeTolerations() []v1.Toleration {
-	return []v1.Toleration{
+	t := []v1.Toleration{
 		{Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
 		{Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoExecute},
+		{Operator: v1.TolerationOpExists, Key: "node.kubernetes.io/not-ready"},
```

Comment:
I think if we only need this when running on DockerEE then we should only tolerate it when that is the Provider.

Reply:
K8s will automatically add this and the 'unreachable' toleration (Effect: NoExecute) for Daemonsets, but since we are setting tolerations explicitly, I don't think K8s will automatically add them. Perhaps this isn't correct, but then I don't understand why it didn't get automatically added in this case. Since we are already adding tolerations explicitly, I think we should go ahead and add both. Also, I don't have any evidence of efficacy (docs or otherwise) for the tolerations we've added for NoSchedule/NoExecute without Keys. What do they do?